src/cmd/kssh/kssh.go

```diff
@@ -257,20 +280,52 @@ func provisionNewKey(config kssh.ConfigFile, keyPath string) error {
 	return nil
 }
 
-// Run SSH with the given key. Calls os.Exit if SSH returns
+// Run SSH with the given key. Calls os.Exit and does not return.
 func runSSHWithKey(keyPath string, remainingArgs []string) {
-	fmt.Printf("\n") // a new line to separate kssh output from ssh output
+	// Create a config file for the default user if necessary
+	useConfig, err := kssh.CreateDefaultUserConfigFile()
+	if err != nil {
+		fmt.Printf("Failed to set default user: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Add the key to the ssh-agent in case we are doing multiple connections (eg via the `-J` flag)
+	err = kssh.AddKeyToSSHAgent(keyPath)
+	if err != nil {
+		fmt.Printf("Failed to add SSH key to the SSH agent: %v\n", err)
+		os.Exit(1)
+	}
+
+	// A new line to separate kssh output from ssh output
+	fmt.Printf("\n")
+
 	argumentList := []string{"-i", keyPath, "-o", "IdentitiesOnly=yes"}
+	if useConfig {
+		warnAboutAlternateConfig(remainingArgs)
+		argumentList = append(argumentList, "-F", kssh.AlternateSSHConfigFile)
+	}
+
 	argumentList = append(argumentList, remainingArgs...)
 
 	cmd := exec.Command("ssh", argumentList...)
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 	cmd.Stdin = os.Stdin
-	err := cmd.Run()
+	err = cmd.Run()
+
 	if err != nil {
 		fmt.Printf("SSH exited with err: %v\n", err)
 		os.Exit(1)
 	}
 	os.Exit(0)
 }
+
+func warnAboutAlternateConfig(arguments []string) {
+	for _, arg := range arguments {
+		if arg == "-F" {
+			fmt.Println("Warning: kssh uses the -F argument in order to implement support for default SSH users. It " +
+				"is not supported to run kssh with a default user and the -F flag. Either do not use the -F flag or run" +
+				"`kssh --clear-default-user` to reset the default SSH user. ")
```

Comment:
i don't love this warning message. lemme see if i understand this. this is firing because the kssh user has _both_ specified the username of the ssh user they'd like to use (via config) _AND_ they've chosen to delegate the username to the running CA bot (by using kssh at all). and they can't do both of those things. so they should either run `--clear-default-user` so they aren't specifying it locally OR they should... what's the alternate path? not using kssh?

Reply:
does this thing displaying mean that your ssh attempt is definitely going to fail? it looks like it's still running.

Reply:
This warning means that the user has run:

```
kssh --set-default-user foo
kssh -F ~/my-ssh-config server
```

This error will not happen if the user does not set a default user (I think most people will not use this option) and will not happen if the user does not set an alternate SSH config (I think most people will just use the default `~/.ssh/config`. Ultimately, this warning message should happen very rarely but it has to happen because we are using a somewhat leaky abstraction for setting the default user. 

The alternate path is not specifying a custom ssh config file (and just using `~/.ssh/config`). 

And no, it may work fine. Or it may not. So I figured it was reasonable to let it run with a warning but happy to change it out. Just undefined behavior.

Reply:
cool. so couple little suggestions. the design of this is a little awkward. i suggest changing this to something like (and this is the caller): 
```go
checkAndWarnOnUnspecifiedBehavior(useConfig, remainingArgs)
if useConfig {
    argumentList = append(argumentList, "-F", kssh.AlternateSSHConfigFile)
}
```

and minor tweak request to the warning message:
```go
fmt.Println("Warning: You passed a -F flag, but kssh also uses this argument in order to implement support for a default SSH username, which you're also using. Either do not use the -F flag or run" +
"`kssh --clear-default-user` to reset the default SSH user and delegate this to the running CA bot.")
```

Reply:
Done.